src/batched/KokkosBatched_Gemm_Handle.hpp

```diff
@@ -0,0 +1,180 @@
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+//
+// Created by Harvey, Evan on 7/13/21.
+//
+
+#ifndef KOKKOSKERNELS_KOKKOSBATCHED_GEMM_HANDLE_HPP
+#define KOKKOSKERNELS_KOKKOSBATCHED_GEMM_HANDLE_HPP
+
+#include "KokkosBatched_Kernel_Handle.hpp"
+
+namespace KokkosBatched {
+
+/// \brief Tpl algorithm types. See BatchedGemmHandle for details.
+namespace GemmTplAlgos {
+enum GEMM_TPL_ALGOS : int { CUBLAS = N_BASE_ALGOS, MAGMA, N };
+}
+
+/// \brief KokkosBatched algorithm types. See BatchedGemmHandle for details.
+namespace GemmKokkosBatchedAlgos {
+enum GEMM_KOKKOS_BATCHED_ALGOS : int {
+  KK_TEAM = GemmTplAlgos::N,
+  KK_TEAMVECTOR,
+  KK_SERIALSIMD,
+  KK_TEAMSIMD,
+  KK_SERIAL_OPT2,
+  KK_TEAMVECTOR_SHMEM,
+  KK_TEAMVECTOR_DBLBUF,
+  N
+};
+}
+
+// clang-format off
+/// \brief Handle for selecting runtime behavior of the BatchedGemm interface.
+///
+/// \var kernelAlgoType  Specifies which algorithm to use for invocation (default, SQUARE).
+///
+///                    Specifies whether to select optimal invocations based on inputs and
+///                    heuristics:
+///                      SQUARE select invocations based on square matrix heuristics where M=N
+///                      TALL   select invocations based on tall   matrix heuristics where M>N
+///                      WIDE   select invocations based on wide   matrix heuristics where M<N
+///                    Note: If the heuristics indicate SIMD views are required for optimal
+///                    performance, notify the user that SIMD views are required for
+///                    optimal performance.
+///
+///                    Specifies which cmake-enabled tpl algorithm to invoke:
+///                      ARMPL    Invoke the ArmPL TPL interface
+///                      MKL      Invoke the MKL TPL interface
+///                      CUBLAS   Invoke the CuBLAS TPL interface
+///                      MAGMA    Invoke the Magma TPL interface
+///                    Note: Requires that input views for A, B, and C reside on either host
+///                    or device depending on the TPL selected.
+///                    Note: If the user selects a TPL, an error will be thrown if:
+///                       1. The TPL is not enabled via cmake
+///                       2. The input views do not reside on the host/device as needed
+///
+///                    Specifies which kokkos-kernels (KK) algorithm to invoke:
+///                      KK_SERIAL            Invoke SerialFUNC     via RangePolicy(BatchSz)
```

Comment:
I feel ```KK_SERIAL``` might be misleading. From the user's point of view, it might be understood as a serial implementation. Should we consider renaming it to something more descriptive, maybe something related to rangepolicy?

Reply:
I think it's OK as is; we note that `RangePolicy` is used here and follow a consistent nomenclature of `LIB_ALGO`.

Reply:
It is just a minor concern. It is fine to me now as long as it is documented.